docs: rephrase plugin configuration in docs (#6557)
python-poetry/poetry: docs/pyproject.md

## `plugins`

> Poetry supports arbitrary plugins which work similarly to `setuptools` entry points.

Reviewer: ```suggestion
Poetry supports arbitrary plugins, which are exposed as the ecosystem-standard [entry points](https://packaging.python.org/en/latest/specifications/entry-points/) and discoverable using `importlib.metadata`. This is similar to (and compatible with) the entry points feature of `setuptools`.
```

The syntax for registering a plugin is:

```toml
[tool.poetry.plugins] # Optional super table

[tool.poetry.plugins."A"]
"B" = "C:D"
```
Which are:

- `A` - type of the plugin, for example `poetry.plugin` or `flake8.extension`
- `B` - name of the plugin

> - `C` - pythonic path to module containing plugin class

Reviewer: ```suggestion
- `C` - python module import path
```

> - `D` - name of plugin class

Reviewer: ```suggestion
- `D` - the entry point of the plugin (a function or class)
```

> - `D` - the entry point of the plugin (a function or class)

Reviewer 2: IMO, having an example usage additionally to the template and details would give even more insights to user.

What about showcasing our own export plugin, as a real life example?
```toml
[tool.poetry.plugins."poetry.application.plugin"]
export = "poetry_plugin_export.plugins:ExportApplicationPlugin"
```

Author: Good idea, I have added the example.
